### DISCUSSION TITLE
A question about disabled LTO for Maturin itself

### DISCUSSION THREAD

**Discussion by zamazan4ik** [PyO3/maturin#2337] [Q&A] [answered]
Hi!

As far as I understand, earlier the Maturin binary was optimized with LTO but since https://github.com/PyO3/maturin/discussions/1012#discussioncomment-3168053 it was disabled due to a bug (if I am wrong here - please correct me).

Is it still an issue or we can re-enable LTO once again for Maturin? According to my local tests (the latest Maturin, Fedora 41, Rust 1.82), enabling `lto = true` for the Release profile at least reduces the binary size from 23 Mib to 19 Mib (didn't perform performance benchmarks though but I think the performance would be at least at the same level).

Thank you.

P.S. Please don't mess it up with the https://github.com/PyO3/maturin/issues/1529 issue - here I am asking about enabling LTO for Maturin itself, not for the Rust-based Python packages built with Maturin.

**Comment by messense** [comment, marked as answer]
Enabled in release workflow for all targets except ppc64le: https://github.com/PyO3/maturin/actions/runs/12094076123/job/33725734878

```
error: failed to get bitcode from object file for LTO (could not find requested section)

error: could not compile `maturin` (bin "maturin") due to 1 previous error
```

**Reply by zamazan4ik** [reply]
Thank you for enabling LTO!

It's kinda interesting that LTO fails for the ppc64le target. I have made a quick research in the upstream and found similar issues:
* https://github.com/rust-lang/rust/issues/129888
* https://github.com/rust-lang/rust/issues/94232

I will report your finding to the upstream to the issues above
